File: src/utils/output.rs
use colored::Colorize;
use tabled::{Table, settings::{Style, themes::Colorization, Color}};
use tabled::settings::object::Rows;

/// Format data as a pretty table or JSON based on output preference
pub fn print_table<T: tabled::Tabled>(items: &[T], output_json: bool) -> String
where
    T: serde::Serialize,
{
    if output_json {
        serde_json::to_string_pretty(items).unwrap_or_else(|_| "[]".to_string())
    } else if items.is_empty() {
        format!("  {}", "No results found.".dimmed())
    } else {
        let mut table = Table::new(items);
        table.with(Style::rounded());
        table.with(Colorization::exact([Color::new("\x1b[1;36m", "\x1b[0m")], Rows::first()));
        table.to_string()
    }
}

/// Print a single item as pretty JSON or a key-value display
pub fn print_detail<T: serde::Serialize>(item: &T, output_json: bool) -> String {
    if output_json {
        serde_json::to_string_pretty(item).unwrap_or_else(|_| "{}".to_string())
    } else {
        // Use colored JSON for non-json output too (looks nice)
        let json = serde_json::to_string_pretty(item).unwrap_or_else(|_| "{}".to_string());
        colorize_json(&json)
    }
}

/// Colorize a JSON string for terminal output
fn colorize_json(json: &str) -> String {
    let mut result = String::new();
    for line in json.lines() {
        let colored_line = if line.contains(':') {
            let parts: Vec<&str> = line.splitn(2, ':').collect();
            if parts.len() == 2 {
                format!("{}: {}", parts[0].cyan(), parts[1])
            } else {
                line.to_string()
            }
        } else {
            line.dimmed().to_string()
        };
        result.push_str(&colored_line);
        result.push('\n');
    }
    result
}

/// Print a success message with a checkmark
pub fn success(msg: &str) {
    println!("  {} {}", "✓".green().bold(), msg);
}

/// Print an error message
pub fn error(msg: &str) {
    eprintln!("  {} {}", "✗".red().bold(), msg);
}

/// Print a warning message
pub fn warning(msg: &str) {
    eprintln!("  {} {}", "⚠".yellow().bold(), msg);
}

/// Print an info message
pub fn info(msg: &str) {
    println!("  {} {}", "ℹ".blue().bold(), msg);
}

/// Status badge with color based on status string
pub fn status_badge(status: &str) -> String {
    match status.to_lowercase().as_str() {
        "active" | "published" | "paid" | "finalized" => format!("{}", status.green().bold()),
        "draft" | "pending" => format!("{}", status.yellow()),
        "cancelled" | "canceled" | "void" | "voided" | "inactive" => format!("{}", status.red()),
        "trialing" | "paused" => format!("{}", status.blue()),
        _ => status.to_string(),
    }
}

/// The FlexPrice ASCII art banner
pub fn print_banner() {
    let banner = r#"
    ╔═══════════════════════════════════════════╗
    ║                                           ║
    ║   ⚡ FlexPrice CLI                        ║
    ║   Usage-based billing, made simple.       ║
    ║                                           ║
    ╚═══════════════════════════════════════════╝
    "#;
    println!("{}", banner.cyan());
}

File: npm/src/utils/output.ts
import chalk from 'chalk';
import Table from 'cli-table3';

export function printTable(rows: Record<string, string>[], outputJson: boolean): string {
  if (outputJson) {
    return `${JSON.stringify(rows, null, 2)}\n`;
  }
  if (rows.length === 0) {
    return `  ${chalk.dim('No results found.')}\n`;
  }
  const cols = Object.keys(rows[0] ?? {});
  const table = new Table({
    head: cols.map((c) => chalk.bold.cyan(c)),
    style: { head: [] },
  });
  for (const row of rows) {
    table.push(cols.map((c) => row[c] ?? ''));
  }
  return `${table.toString()}\n`;
}

export function printDetail(item: unknown, outputJson: boolean): string {
  if (outputJson) {
    return `${JSON.stringify(item, null, 2)}\n`;
  }
  const json = JSON.stringify(item, null, 2);
  return colorizeJson(json);
}

function colorizeJson(json: string): string {
  let result = '';
  for (const line of json.split('\n')) {
    if (line.includes(':')) {
      const idx = line.indexOf(':');
      const key = line.slice(0, idx);
      const rest = line.slice(idx + 1);
      result += `${chalk.cyan(key)}:${rest}\n`;
    } else {
      result += `${chalk.dim(line)}\n`;
    }
  }
  return result;
}

export function success(msg: string): void {
  console.log(`  ${chalk.green.bold('✓')} ${msg}`);
}

export function error(msg: string): void {
  console.error(`  ${chalk.red.bold('✗')} ${msg}`);
}

export function warning(msg: string): void {
  console.error(`  ${chalk.yellow.bold('⚠')} ${msg}`);
}

export function info(msg: string): void {
  console.log(`  ${chalk.blue.bold('ℹ')} ${msg}`);
}

export function statusBadge(status: string): string {
  const s = status.toLowerCase();
  if (['active', 'published', 'paid', 'finalized'].includes(s)) {
    return chalk.green.bold(status);
  }
  if (['draft', 'pending'].includes(s)) {
    return chalk.yellow(status);
  }
  if (['cancelled', 'canceled', 'void', 'voided', 'inactive'].includes(s)) {
    return chalk.red(status);
  }
  if (['trialing', 'paused'].includes(s)) {
    return chalk.blue(status);
  }
  return status;
}
